src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala

```diff
@@ -35,16 +37,66 @@ import scala.util.{ Failure, Success, Try }
  *
  * @param bag the bag containing the data files
  */
-case class DataFiles(bag: DansBag) extends DebugEnhancedLogging {
+case class DataFiles(bag: DansBag, bagId: UUID) extends DebugEnhancedLogging {
+  private lazy val uploadRoot = {
```

Comment:
Would you please add a comment briefly explaining the reason for having both these options: a.k.a. old situation vs. new situation. Future generations of developers will thank you later 😉